tests/Unit/ControlSystem/Test_PiecewisePolynomial.cpp

```diff
@@ -112,6 +90,112 @@ SPECTRE_TEST_CASE(
   CHECK(approx(lambdas2[0][0]) == 1.0);
   CHECK(approx(lambdas2[0][1]) == 1.0);
 }
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial",
+                  "[ControlSystem][Unit]") {
+  PUPable_reg(FunctionsOfTime::PiecewisePolynomial<2>);
+  PUPable_reg(FunctionsOfTime::PiecewisePolynomial<3>);
+
+  {
+    INFO("Core test");
+    double t = 0.0;
+    const double dt = 0.6;
+    const double final_time = 4.0;
+    constexpr size_t deriv_order = 3;
+
+    // test two component system (x**3 and x**2)
+    const std::array<DataVector, deriv_order + 1> init_func{
+        {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+    {
+      INFO("Test with simple construction.");
+      FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+      FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t2 =
+          serialize_and_deserialize(f_of_t);
+
+      test(make_not_null(&f_of_t), make_not_null(&f_of_t), t, dt, final_time);
+      test(make_not_null(&f_of_t2), make_not_null(&f_of_t2), t, dt, final_time);
```

Comment:
I would find this test (and the others) much cleaner if instead of checking that the two `f_of_t`'s behave the same, you checked that they were actually identical using an `operator==`. Isn't this how we test serialization in most of the code? I find that separating the test of the behavior from the test of the serialization helps modularity and understandability of the test.

Is there a particular reason why a `FunctionOfTime` shouldn't have an `operator==`?

Reply:
The derived classes could have `operator==`, though they're not copyable so it doesn't really make sense to have them comparable. Additionally, we've had a number of times where `operator==` was implemented incorrectly, so the behavioral test is value in either case.

Reply:
Okay, sounds like reason enough 👍